infrahub_sdk/yaml.py

```diff
@@ -23,3 +46,52 @@ def load_content(self) -> None:
         if not self.content:
             self.error_message = "Empty YAML/JSON file"
             self.valid = False
+
+    def validate_content(self) -> None:
+        pass
+
+    @classmethod
+    def load_from_disk(cls, paths: list[Path]) -> list[Self]:
```

Comment:
We currently read most of the data from disk as a sync function, i.e. the jinja2 templates when used from within the git-agent and also even the loading of .infrahub.yml files. I see this being used within the git-agents as well. Even though we don't have the file loaders as async now I'm wondering if this should be an async function from the start, or if we just introduce a new one that is async once we actually read the files from disk in an async way.

Reply:
Since it's sync for now, my vote is to keep it as a sync function but it would be good to start using `anyio.Path` to properly support Async for all disk access. I'll open an issue to track that